Signal/src/ViewControllers/CountryCodeViewController.m

```diff
@@ -2,99 +2,159 @@
-    self.searchBar.delegate = self;
-    if (self.shouldDismissWithoutSegue) {
-        self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemStop
-                                                                                              target:self
-                                                                                                       action:@selector(dismissWasPressed:)];
-    }
+    self.title = NSLocalizedString(@"COUNTRYCODE_SELECT_TITLE", @"");
+
+    self.countryCodes = [PhoneNumberUtil countryCodesForSearchTerm:nil];
+
+    self.navigationItem.leftBarButtonItem =
+        [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemStop
+                                                      target:self
+                                                      action:@selector(dismissWasPressed:)];
+
+    [self createViews];
 }
 
-#pragma mark - UITableViewDelegate
+- (void)createViews
+{
+    // Search
+    UISearchBar *searchBar = [UISearchBar new];
+    _searchBar = searchBar;
+    searchBar.searchBarStyle = UISearchBarStyleMinimal;
+    searchBar.delegate = self;
+    searchBar.placeholder = NSLocalizedString(@"SEARCH_BYNAMEORNUMBER_PLACEHOLDER_TEXT", @"");
+    searchBar.backgroundColor = [UIColor whiteColor];
+    [searchBar sizeToFit];
+
+    self.tableView.tableHeaderView = searchBar;
 
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-    return (NSInteger)_countryCodes.count;
+    [self updateTableContents];
 }
 
-- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
-    CountryCodeTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CONTRY_CODE_TABLE_CELL_IDENTIFIER];
-    if (!cell) {
-        cell = [[CountryCodeTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault
-                                               reuseIdentifier:CONTRY_CODE_TABLE_CELL_IDENTIFIER];
+#pragma mark - Table Contents
+
+- (void)updateTableContents
+{
+    OWSTableContents *contents = [OWSTableContents new];
+
+    __weak CountryCodeViewController *weakSelf = self;
+    OWSTableSection *section = [OWSTableSection new];
+
+    for (NSString *countryCode in self.countryCodes) {
+        OWSAssert(countryCode.length > 0);
+        OWSAssert([PhoneNumberUtil countryNameFromCountryCode:countryCode].length > 0);
+        OWSAssert([PhoneNumberUtil callingCodeFromCountryCode:countryCode].length > 0);
+        OWSAssert(![[PhoneNumberUtil callingCodeFromCountryCode:countryCode] isEqualToString:@"+0"]);
+
+        [section addItem:[OWSTableItem itemWithCustomCellBlock:^{
+            UITableViewCell *cell = [UITableViewCell new];
+            cell.textLabel.text = [PhoneNumberUtil countryNameFromCountryCode:countryCode];
+            cell.textLabel.font = [UIFont ows_regularFontWithSize:18.f];
```

Comment:
This view used to use medium but our other similar views use regular, so I switched.